### PR TITLE
HGNCParser

### PR DESCRIPTION
**Pull request by tgrego**
## Description

Porting of the HGNC parser to the new repo.

## Testing

Tests are included

### CODE REVIEW

**Comment by premanand17** [comment]
Please check the failing test.

Can't locate object method "add_synonyms_for_hgnc_vgnc" via package "Bio::EnsEMBL::Xref::DBSQL::BaseAdaptor" at /home/travis/build/Ensembl/ensembl-xref/modules/Bio/EnsEMBL/Xref/Parser/HGNCParser.pm line 314, <$fh> line 2.
Tests were run but no plan was declared and done_testing() was not seen.
Looks like your test exited with 255 just after 3.
modules/t/hgnc.t ..................

File: modules/Bio/EnsEMBL/Xref/Parser/HGNCParser.pm [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+# Approved symbol
+# Approved name
+# Previous symbols
+# Synonyms
+# NCBI Gene ID
+# Ensembl gene ID
+# RefSeq IDs
+# CCDS IDs
+# Locus specific databases
+
```
Comment (premanand17) [inline modules/Bio/EnsEMBL/Xref/Parser/HGNCParser.pm]: Column names already listed in POD. Repetition!!

File: modules/Bio/EnsEMBL/Xref/Parser/HGNCParser.pm [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+  my $sql =(<<'CCDS');
+  SELECT ta.value, t.stable_id
+  FROM transcript t
+  INNER JOIN transcript_attrib ta ON t.transcript_id = ta.transcript_id
+  INNER JOIN attrib_type a ON ta.attrib_type_id = a.attrib_type_id
+  WHERE a.code = 'ccds_transcript';
+CCDS
+
+  my $sth = $core_dba->dbc->prepare($sql);
+  $sth->execute() or croak( $core_dba->dbc->errstr() );
```
Comment (premanand17) [inline modules/Bio/EnsEMBL/Xref/Parser/HGNCParser.pm]: confess!!

File: modules/Bio/EnsEMBL/Xref/Parser/HGNCParser.pm [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+# CCDS IDs
+# Locus specific databases
+
+
+  my $input_file = Text::CSV->new({
+    sep_char       => "\t",
+    empty_is_undef => 1,
+    binary         => 1,
+    auto_diag      => 1
+  }) or croak "Cannot use file $file: ".Text::CSV->error_diag ();
```
Comment (premanand17) [inline modules/Bio/EnsEMBL/Xref/Parser/HGNCParser.pm]: confess!

File: modules/Bio/EnsEMBL/Xref/Parser/HGNCParser.pm [hunk 4]
```diff
@@ -0,0 +1,398 @@
...
+        source_id  => $self->{source_ids}->{'ccds'},
+        name       => $acc,
+        species_id => $self->{species_id},
+        dead       => $previous_symbols,
+        alias      => $synonyms
+    });
+    $self->{name_count}->{ccds}++;
+  }
+
+  # Direct LRG to ENST mappings
```
Comment (premanand17) [inline modules/Bio/EnsEMBL/Xref/Parser/HGNCParser.pm]: ENST or ENSG?!

File: modules/Bio/EnsEMBL/Xref/Parser/HGNCParser.pm [hunk 5]
```diff
@@ -0,0 +1,398 @@
...
+            source_id      => $self->{source_ids}->{'refseq_manual'},
+            species_id     => $self->{species_id}
+        });
+        $self->{name_count}->{refseq_manual}++;
+      }
+
+      $self->{xref_dba}->add_synonyms_for_hgnc_vgnc({
+          source_id  => $self->{source_id},
+          name       => $acc,
+          species_id => $self->{source_ids}->{'refseq_manual'},
```
Comment (premanand17) [inline modules/Bio/EnsEMBL/Xref/Parser/HGNCParser.pm]: Check the key. species_id or source_id??

Comment (premanand17) [reply modules/Bio/EnsEMBL/Xref/Parser/HGNCParser.pm]: Perhaps the value should be $self->{species_id}

**Comment by tgrego** [comment]
That failing test is due a requirement of VGNCParser PR to be merged in.
Will be gone as soon as that code is merged.

File: modules/Bio/EnsEMBL/Xref/Parser/HGNCParser.pm (hunk 5 above)
Comment (tgrego) [reply modules/Bio/EnsEMBL/Xref/Parser/HGNCParser.pm]: good catch

**Pull request merged**